app/src/main/java/com/asfoundation/wallet/ui/transact/TransferFragment.kt

```diff
@@ -48,88 +32,28 @@ class TransferFragment : BasePageViewFragment(), TransferFragmentView {
+  lateinit var presenter: TransferPresenter
 
-  lateinit var navigator: TransactNavigator
-  private lateinit var activityResultSharer: ActivityResultSharer
   private lateinit var doneClick: PublishSubject<Any>
   private lateinit var qrCodeResult: BehaviorSubject<Barcode>
-  private var disposable: Disposable? = null
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     doneClick = PublishSubject.create()
     qrCodeResult = BehaviorSubject.create()
-    disposable =
-        confirmationRouter.transactionResult
-            .doOnNext { activity?.onBackPressed() }
-            .subscribe()
-    presenter = TransferPresenter(this, CompositeDisposable(), CompositeDisposable(), interactor,
-        Schedulers.io(), AndroidSchedulers.mainThread(), findDefaultWalletInteract,
-        walletBlockedInteract, context!!.packageName, formatter)
-  }
-
-  override fun openEthConfirmationView(walletAddress: String, toWalletAddress: String,
-                                       amount: BigDecimal): Completable {
-    return Completable.fromAction {
-      val transaction = TransactionBuilder(TokenInfo(null, "Ethereum", "ETH", 18))
-      transaction.amount(amount)
-      transaction.toAddress(toWalletAddress)
-      transaction.fromAddress(walletAddress)
-      confirmationRouter.open(activity, transaction)
-    }
-  }
-
-  override fun openAppcConfirmationView(walletAddress: String, toWalletAddress: String,
-                                        amount: BigDecimal): Completable {
-
-    return defaultTokenInfoProvider.defaultToken.doOnSuccess {
-      with(TransactionBuilder(it)) {
-        amount(amount)
-        toAddress(toWalletAddress)
-        fromAddress(walletAddress)
-        confirmationRouter.open(activity, this)
-      }
-    }
-        .ignoreElement()
-  }
-
-  override fun openAppcCreditsConfirmationView(walletAddress: String,
-                                               amount: BigDecimal,
-                                               currency: TransferFragmentView.Currency): Completable {
-    return Completable.fromAction {
-      val currencyName = when (currency) {
-        TransferFragmentView.Currency.APPC_C -> getString(R.string.p2p_send_currency_appc_c)
-        TransferFragmentView.Currency.APPC -> getString(R.string.p2p_send_currency_appc)
-        TransferFragmentView.Currency.ETH -> getString(R.string.p2p_send_currency_eth)
-      }
-      navigator.openAppcoinsCreditsSuccess(walletAddress, amount, currencyName)
-    }
   }
 
   override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                             savedInstanceState: Bundle?): View? {
     return inflater.inflate(R.layout.transact_fragment_layout, container, false)
   }
 
+  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    super.onActivityResult(requestCode, resultCode, data)
+    presenter.onActivityResult(requestCode, resultCode, data)
+  }
+
```

Comment:
Why is this between onCreateView and onViewCreated?
Isn't there a usual order for these fragment methods?
Move onActivityResult below onViewCreated, and review order.